core/src/main/scala/org/apache/spark/ui/jobs/JobsTab.scala

```diff
@@ -35,4 +37,18 @@ private[ui] class JobsTab(parent: SparkUI) extends SparkUITab(parent, "jobs") {
 
   attachPage(new AllJobsPage(this))
   attachPage(new JobPage(this))
+
+  def handleKillRequest(request: HttpServletRequest): Unit = {
+    if (killEnabled && (parent.securityManager.checkModifyPermissions(request.getRemoteUser))) {
+      val killFlag = Option(request.getParameter("terminate")).getOrElse("false").toBoolean
+      val jobId = Option(request.getParameter("id")).getOrElse("-1").toInt
+      if (jobId >= 0 && killFlag && jobProgresslistener.activeJobs.contains(jobId)) {
+        sc.get.cancelJob(jobId)
+      }
```

Comment:
Creating an `Option` only to immediately `get` the value out of it is poor style, and unnecessary.
```scala
val jobId = Option(request.getParameter("id"))
jobId.foreach { id =>
  if (killFlag && jobProgresslistener.activeJobs.contains(id)) {
    sc.get.cancelJob(id)
  }
}
```